splinterd/src/config/env.rs

```diff
@@ -44,23 +44,69 @@ const METRICS_USERNAME_ENV: &str = "SPLINTER_INFLUX_USERNAME";
 #[cfg(feature = "tap")]
 const METRICS_PASSWORD_ENV: &str = "SPLINTER_INFLUX_PASSWORD";
 
-pub struct EnvPartialConfigBuilder;
+/// Trait that outlines a basic read-only environment variable store
+pub trait EnvStore {
+    /// Returns an environment variable for the given key. Results are intended to mirror the
+    /// functionality of `env::var(key)`.
+    ///
+    /// # Arguments
+    ///
+    /// * `key` - A string slice that holds the name of the environment variable
+    ///
+    fn get(&self, key: &str) -> Result<String, VarError>;
```

Comment:
Given we never use the error, should this just return `Option<String>`?

Reply:
Updated!